### PR TITLE
DM-22081: Add NB to compare HiTS 2015 vs 2014 templates

### PR DESCRIPTION
**Pull request by isullivan**
Also adds a python module for evaluating a metric for how well constrained the DCR model is for a set of observations.

### CODE REVIEW

File: dcrMetric.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        
+        Returns
+        -------
+        visit_measure: `float`
+            A one dimensional parameterization of airmass and hour angle.
+        """
+        visit_measure = (airmass - 1.)*np.sign(hour_angle)
+        return visit_measure
+    
+    def parameterizeTable(self, visits=None):
```
Comment (mrawls) [inline dcrMetric.py]: Here and throughout, you switch between calling the main pandas dataframe you're using a "table" and a "database." Maybe just call it a dataframe? At the very least, please be consistent.

File: dcrMetric.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+        
+        Parameters
+        ----------
+        visit: `float`
+            The visit identifier of the observation to be removed.
+        """
+        try:
+            airmass = self.table['airmass'][visit]
+            hour_angle = self.table['hour_angle'][visit]
+            self.table.drop(visit, inplace=True)
```
Comment (mrawls) [inline dcrMetric.py]: Do all three of these lines need to be in the try block, or can one or more of them happen before it or in the else block? It's not immediately clear which one you're catching an exception on. (But if it's any/all, it's fine to leave it as is.)

File: dcrMetric.py [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+        Returns
+        -------
+        visit_metric: `float`
+            The metric evaluating how well a new observation can be constrained for the current DCR model.
+        """
+        hour_angle = visitInfo.getBoresightHourAngle().asRadians()
+        airmass = visitInfo.getBoresightAirmass()
+        return self.evaluateVisit(airmass, hour_angle)
+    
+    def plotMetric(self, window=1000, plot_measures=True):
```
Comment (mrawls) [inline dcrMetric.py]: Please give a little more detail in the docstring about what this is plotting. I stared at these plots for a while in the notebook until I worked out that a larger DCR metric is what you're aiming for and the x-axis is your visit measure parameter.

File: dcrMetric.py [hunk 4]
```diff
@@ -0,0 +1,301 @@
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import scipy.stats
+
+
+class DcrMetric:
```
[the remark this answers is not in the document]
Comment (isullivan) [reply dcrMetric.py]: Oops. That's the problem with developing code in a notebook first - there there were a lot of code style errors and missing docs. I've updated it throughout.

File: dcrMetric.py [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+            The observed airmass of the observation.
+        hour_angle: `float`
+            The observed hour angle of the observation.
+        """
+        if visit in self.table.index:
+            raise KeyError(f"Visit {visit} is already present in the table, and cannot be added. Skipping it.")
+        self.table.loc[visit] = [airmass, hour_angle]
+        self.kde = None
+        
+    def addVisitInfo(self, visitInfo):
```
[the remark this answers is not in the document]
Comment (isullivan) [reply dcrMetric.py]: Yes, `visitInfo` should remain unchanged in Gen 3.

[the remark this answers is not in the document]
Comment (timj) [reply dcrMetric.py]: VisitInfo is butler-agnostic. We might add information to it but it's not going away.

**Pull request merged**